### ISSUE TITLE
Vim9: trying to redefine a function-local function is silently ignored

### ISSUE THREAD

**Issue by lacygoill** [vim/vim#6583]
**Describe the bug**

Trying to redefine a function-local function is silently ignored.

**To Reproduce**

Run this shell command:

    vim -Nu NONE -S <(cat <<'EOF'
        vim9script
        def Outer()
            def Inner()
                echo 'first'
            enddef
            def Inner()
                echo 'second'
            enddef
            Inner()
        enddef
        Outer()
    EOF
    )

`Outer()` outputs `first`.

**Expected behavior**

`Outer()` raises an error when it tries to redefine the `Inner()` function.

**Environment**

 - Vim version: 8.2 Included patches: 1-1328
 - OS: Ubuntu 16.04.6 LTS
 - Terminal: XTerm(358)

**Additional context**

The same issue applies to a function local to a block:
```vim
vim9script
def Outer()
    if 1
        def Inner()
            echo 'first'
        enddef
        def Inner()
            echo 'second'
        enddef
        Inner()
    endif
enddef
Outer()
```
    first

And to an imported function:
```vim
vim9script

mkdir('/tmp/import', 'p')

let lines =<< trim END
    vim9script
    export def Func()
        echo 'first'
    enddef
END
writefile(lines, '/tmp/import/a.vim')

lines =<< trim END
    vim9script
    export def Func()
        echo 'second'
    enddef
END
writefile(lines, '/tmp/import/b.vim')

set rtp+=/tmp
import Func from 'a.vim'
import Func from 'b.vim'
Func()
```
    first

---

This is inconsistent with functions which are local to a script:
```vim
vim9script
def Func()
    echo 'first'
enddef
def Func()
    echo 'second'
enddef
```
    E122: Function <SNR>1_Func already exists, add ! to replace it

Notice how an error message is correctly raised ([although a bit misleading](https://github.com/vim/vim/issues/6581)).

**Comment by lacygoill** [comment]
The issue is specific to `:def` functions inside a `:def` function.

It does not affect `:fu` functions inside a `:fu` function:
```vim
vim9script
fu Outer()
    fu Inner()
        echo 'first'
    endfu
    fu Inner()
        echo 'second'
    endfu
    call Inner()
endfu
Outer()
```
    E122: Function Inner already exists, add ! to replace it

And it does not affect `:def` functions inside a `:fu` function:
```vim
vim9script
fu Outer()
    def Inner()
        echo 'first'
    enddef
    def Inner()
        echo 'second'
    enddef
    call Inner()
endfu
Outer()
```
    E122: Function Inner already exists, add ! to replace it

**Comment by lacygoill** [comment]
Fixed by [8.2.1349](https://github.com/vim/vim/releases/tag/v8.2.1349).
